CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+## 0.12.2
+
+* Fixed `unorderedMatches` in cases where the matchers may match more than one
```

Comment:
I could not find any breakages related to this change in a global presubmit. I scanned through the few failures and they all looked like flakes.

We do have other options though. I am fixing this because I want the correct semantics for `containsAll` - https://github.com/dart-lang/matcher/issues/72, but we could also implement the correct semantics there and deprecate this broken one.

Once we have `containsAll` then `unorderedMatches` (and by extension `unorderedEquals`) become pretty much unnecessary because these are equivalent:

```dart
expect(something, unorderedEquals(expected);

// can also be written as:
expect(something.length, expected.length);
expect(something, containsAll(expected)));
```

And `unorderedEquals` could always be substituted with `unorderedMatches` as long as none of the values were already matchers.

Reply:
If we don't see large numbers of breakages internally then I say its safe to land :+1: